src/test/java/net/snowflake/ingest/TestUtils.java

```diff
@@ -317,4 +322,27 @@ public static SimpleIngestManager getManagerUsingBuilderPattern(
         .setUserAgentSuffix(userAgentSuffix)
         .build();
   }
+
+  /**
+   * Given a channel and expected offset, this method waits until the last committed offset is equal
+   * to the passed offset
+   */
+  public static void waitForOffset(SnowflakeStreamingIngestChannel channel, String expectedOffset)
+      throws InterruptedException {
+    int counter = 0;
+    String lastCommittedOffset = null;
+    while (counter < 600) {
+      String currentOffset = channel.getLatestCommittedOffsetToken();
+      if (Objects.equals(expectedOffset, currentOffset)) {
+        return;
+      }
+      lastCommittedOffset = currentOffset;
+      counter++;
+      Thread.sleep(100);
```

Comment:
We might want to bump this up to a full second

Reply:
Increasing the sleep to 1 second would prolog the test run significantly. This method is called many times and the commit is usually processed quickly, so waiting for 1 second in each iteration is unnecessary.